lib/engine/game/g_1849/game.rb

```diff
@@ -771,12 +824,137 @@ def bank_sort(corporations)
         end
 
         def player_value(player)
-          player.value - @player_debts[player]
+          player.value - @player_debts[player] - player.shares_by_corporation.sum do |corp, _|
+            player.num_shares_of(corp) * corp.loans.size * 100
+          end
         end
 
         def par_prices
           @stock_market.share_prices_with_types(@available_par_groups)
         end
+
+        # code below is for variant rules
+
+        def acquiring_station_tokens?
+          @acquiring_station_tokens ||= @optional_rules&.include?(:acquiring_station_tokens)
+        end
+
+        def electric_dreams?
+          @electric_dreams ||= @optional_rules&.include?(:electric_dreams)
+        end
+
+        def bonds?
+          @bonds ||= @optional_rules&.include?(:bonds)
+        end
+
+        def reduced_4p_corps?
+          @reduced_4p_corps ||= @optional_rules&.include?(:reduced_4p_corps)
+        end
+
+        def event_buy_tokens!
+          @log << "-- Event: #{EVENTS_TEXT[:buy_tokens][1]} --"
+        end
+
+        def event_bonds!
+          @log << "-- Event: #{EVENTS_TEXT[:bonds][1]} --"
+        end
+
+        def event_e_tokens!
+          @log << "-- Event: #{EVENTS_TEXT[:e_tokens][1]} --"
+        end
+
+        # code below is for Bonds variant
+        NUM_LOANS = 6
```

Comment:
not needed?

Reply:
If you don't add/subtract loans when they are taken or paid off, then it doesn't matter. Let's remove.

Reply:
done